joomla/joomla-cms
# [4.0] Core spinning logo

Pull Request for Issue # .

### Summary of Changes
Just a custom element that creates a spinning logo.

### Based on the this codePen: https://codepen.io/nightshiftc/pen/kxmrq?q=Joomla&limit=all&type=type-pens

### Testing Instructions
Apply this patch.
Make sure Custom Fields are enabled.
Try to edit an article and change the category
Boom...

Enjoy the infinite loop...

### Preview
![joomla-loader](https://user-images.githubusercontent.com/3889375/36202421-0d405858-1184-11e8-8dec-6a8fd2362a7b.gif)


### RFC part:
#### Still under discussion
Core.js is just a collection of helper methods, this needs to cleaned up. Eg submit button or submit form is not required unless you already have a form, the spinner is used only in 2-3 pages abut we still load all this things in every page load. This needs to be more efficient, the jQuery days (eg one library has it all) has gone.

#### Resolved: we're going back to self encapsulated js files!
Another issue: Custom elements in their previous iteration had encapsulated the css, meaning one http request and you had everything. When discussing CSP @wilsonge, suggested and we implemented the separation of js and css. This is wrong, once your script is in the allowed list (eg your domain) then whatever they do (if not trying to insert a script or css from another domain, which obviously is not the case here and will never be) is ok. So I'm suggesting to go back to encapsulating everything in the js file as we did before.

### Documentation Changes Required
Yes, although I haven't changed the core.js function this needs to be done and would **NOT** be B/C because even if we add a simple creator for the element there is another  B/C break: you need to manually include the loader element in every page you might want to display it...
